packages/compile-solidity/src/compilerSupplier/loadingStrategies/VersionRange.ts

```diff
@@ -6,7 +6,19 @@ import originalRequire from "original-require";
 
 // must polyfill AbortController to use axios >=0.20.0, <=0.27.2 on node <= v14.x
 import "../../polyfill";
-import { default as axios, AxiosResponse } from "axios";
+import { default as axios_plain, AxiosResponse } from "axios";
+
+var proxy = process.env.https_proxy || process.env.http_proxy;
+var axios;
+if (proxy) {
+  const HttpsProxyAgent = require("https-proxy-agent");
+  const agent = new HttpsProxyAgent(proxy);
+  axios = axios_plain.create({
+    httpsAgent: agent
+  });
+} else {
+  axios = axios_plain;
+}
```

Comment:
This can be removed by initializing `axios` in the declaration statement above.

Reply:
the else clause cannot be removed because the assignments to the variable "axios" carry different types, in one case (proxy) it is an AxiosInstance, in the other case (else) it is an AxiosStatic
removing the else clause leads to a typescript error:
'AxiosInstance' is missing the following properties from type 'AxiosStatic'